### ISSUE TITLE
Improve Compatibility with C++ Applications

### ISSUE THREAD

**Issue by nsi-ltd** [LiamBindle/MQTT-C#93]
When trying to use this code with a C++ application the runtime was failing to find any exported function. On inspection the typical construct that is added to header files was found to be missing. This should be applied to mqtt.h and mqtt_pal.h

```
#ifndef __MQTT_H__
#define __MQTT_H__

#ifdef __cplusplus
extern "C" {
#endif

<rest of code>

#ifdef __cplusplus
}
#endif

#endif
```

**Comment by LiamBindle** [comment]
@nsi-ltd I'd support this. I'm happy to merge a PR with this update :smile:

**Comment by nsi-ltd** [comment]
Hi Liam,

Thank you for your support and efforts in producing this project.

**Comment by LiamBindle** [comment]
Fixed by @markrad in #102
